### PR TITLE
Add a series of infrastructure files from ASLPrep/XCP-D

### PR DESCRIPTION
**Pull request by tsalo**
Closes none.

## Changes proposed in this pull request

- Add issue templates for bug reports and new feature requests. Also add a link to Neurostars for usage questions.
- Add dependabot and linting GitHub Action workflows.
    - The linting job is going to fail for the foreseeable future.
- Move pull request template from `docs/` to `.github/`.
- Add release notes builder template.
    - Remove release-drafter config. Not sure if this conflicts with the release notes builder though. Does it just modify the CHANGES file?
- Add badges to README, including paper DOI, GitHub repo, Docker image, CodeCov, and License.
    - I went on Zenodo and enable the repo, but there won't be a DOI from there until your next release.

### CODE REVIEW

File: .github/workflows/lint.yml
Comment (mattcieslak) [inline .github/workflows/lint.yml:1]: Maybe we should hold off on this one? Getting a linter back in the repo may take a serious amount of rewriting to pass

File: README.rst [hunk 1]
```diff
@@ -3,12 +3,33 @@
...
 
+.. image:: https://img.shields.io/badge/docker-pennbbl/qsiprep-brightgreen.svg?logo=docker&style=flat
+  :target: https://hub.docker.com/r/pennbbl/qsiprep/tags/
+  :alt: Docker
+
 .. image:: https://circleci.com/gh/PennLINC/qsiprep/tree/master.svg?style=svg
   :target: https://circleci.com/gh/PennLINC/qsiprep/tree/master
+  :alt: Test Status
+
+.. image:: https://codecov.io/gh/PennLINC/qsiprep/branch/master/graph/badge.svg
```
Comment (mattcieslak) [inline README.rst:22]: This just says "unknown", maybe we add it after we get codecov up and running?

**Review by mattcieslak — APPROVED** [review]

File: README.rst (hunk 1 above)
Comment (tsalo) [reply README.rst]: Sure. I've removed it for now.

File: .github/workflows/lint.yml
Comment (tsalo) [reply .github/workflows/lint.yml:1]: I just set the target branch to `main_`, which should effectively disable the workflow without deleting the config.

Comment (tsalo) [reply .github/workflows/lint.yml:1]: oh lol the primary branch is `master`, so it doesn't actually have an impact either way

**Comment by tsalo** [comment]
@mattcieslak since I addressed your review comments, is this good to merge?

**Comment by mattcieslak** [comment]
all good!

**Pull request merged**